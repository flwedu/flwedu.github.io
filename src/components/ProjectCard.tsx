export type Project = {
  title: string;
  description: string;
  tags: string[];
  imgUrl: string;
  applicationLink: string | null;
  githubLink: string | null;
};

export function ProjectCard(props: { project: Project }) {
  const { project } = props;

  return (
    <div className="card flex flex-col gap-2 max-w-sm p-2 my-2 rounded-lg shadow-lg space-x-4 items-center place-content-between">
      <div className="card-header flex justify-center">
        <img
          src={project.imgUrl}
          alt="Imagem do projeto"
          className="w-40 rounded-sm"
        />
      </div>
      <div className="card-body">
        <h3>{project.title}</h3>
        <p>{project.description}</p>
      </div>
      <div className="card-footer flex items-center gap-2">
        {project.applicationLink && (
          <ProjectCardButton
            key="1"
            name="Visite App"
            url={project.applicationLink}
          />
        )}
        {project.githubLink && (
          <ProjectCardButton key="2" name="Github" url={project.githubLink} />
        )}
      </div>
    </div>
  );
}

function ProjectCardButton(props: { name: string; url: string }) {
  return (
    <a
      href={props.url}
      target="_blank"
      className="text-center rounded-md px-1 py-2 w-24 bg-mainPurple-500 hover:bg-mainPurple-100"
    >
      {props.name}
    </a>
  );
}
