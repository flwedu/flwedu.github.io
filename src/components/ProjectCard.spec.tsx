import { it, expect, describe } from 'vitest'
import { render } from '../utils/test-utils'
import { ProjectCard } from "./ProjectCard";

describe("Project Card component tests", () => {
  it("Should render a component that matches the snapshot", () => {
    const project = {
      title: "Test App",
      description: "Just a test app",
      tags: ["react"],
      imgUrl: "",
      applicationLink: null,
      githubLink: null,
    };

    const { container } = render(<ProjectCard project={project} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
