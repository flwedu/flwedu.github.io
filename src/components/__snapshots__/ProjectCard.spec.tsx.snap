// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Card component tests Rendered component should match the snapshot 1`] = `
<div
  class="card flex flex-col gap-2 max-w-sm p-2 my-2 rounded-lg shadow-lg space-x-4 items-center place-content-between"
>
  <div
    class="card-header flex justify-center"
  >
    <img
      alt="Imagem do projeto"
      class="w-40 rounded-sm"
      src=""
    />
  </div>
  <div
    class="card-body"
  >
    <h3>
      Test App
    </h3>
    <p>
      Just a test app
    </p>
  </div>
  <div
    class="card-footer flex items-center gap-2"
  />
</div>
`;
